2/homework/laughter_prediction/rnn.py

```diff
@@ -0,0 +1,39 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+
+class LibrosaFeaturesRnn(nn.Module):
+    def __init__(self, fbank_features, mfcc_features, hidden_size, batch_size=1):
+        super(LibrosaFeaturesRnn, self).__init__()
+
+        self.fbank_features = fbank_features
+        self.mfcc_features = mfcc_features
+        self.hidden_size = hidden_size
+        self.batch_size = batch_size
+
+        self.fbank_lstm = nn.LSTM(fbank_features, hidden_size)
+        self.mfcc_lstm  = nn.LSTM(mfcc_features, hidden_size)
+        self.linear_both = nn.Linear(hidden_size * 2, 2)
+        self.linear_mfcc = nn.Linear(hidden_size, 2)
+
+    def init_hidden_state(self, device):
+        return (torch.randn(1, self.batch_size, self.hidden_size).to(device),
```

Comment:
In pytorch you don't need to specify device for particular layers since you can move your whole nn.Module to device all at once